typespec-tests/src/test/java/com/type/model/inheritance/NestedDiscriminatorTests.java

```diff
@@ -39,10 +45,35 @@ void getRecursiveModel() {
     }
 
 
-    @Disabled("Polymorphic deserialization doesn't support multiple levels of inheritance in Jackson, https://github.com/FasterXML/jackson-databind/issues/1188")
     @Test
+    @Disabled("1. The item `kind` is missing in the generated json file by the method `toJson` of the class `Shark`. " +
+            "2. The output order of each item in the generated json file by the method `toJson` of the class `Salmon` is incorrect.")
```

Comment:
Why order affect the test? JSON object does not care about order, isn't it?

Reply:
Fixed in the new version.